### PR TITLE
feat(common): rework and update cnpg code

### PR DESCRIPTION
**Pull request by Ornias1993**
**Description**
This PR aims to rework and update cnpg code to expose more features of CNPG to our users.
Primarily a big change would be the support of more backup and restore options.

- add some missing options for both cluster and pooler
- seperates cluster and pooler sections
- add labels/annotions for pooler and cluster seperatly as well as cnpg-wide labels-annotations
- Simplify storage logic
- Disable metrics by default

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [x] 🪛 Bugfix
- [x] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

It will be very loosely inspired by the work done here, mostly because it gives some nice values.yaml structure on how to deal with the many backup options:
https://github.com/cloudnative-pg/charts/pull/106

It's going to be flagged as "breaking", because of the shear scope and the fact it would likely also include a change in standard operating procedures when it comes to backup-and-restore for our users.

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [ ] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

**Comment by Ornias1993** [comment]
This should now be mostly feature complete afaik.

TODO:
 Only thing missing is the creation of an initial backup, when users enable backups.

**Comment by stavros-k** [comment]
Did some refactoring, will continue later, currently its in an non-working state.
Please dont push any commits until I'm done.

File: library/common/templates/spawner/_cnpg.tpl [hunk 1]
```diff
@@ -23,80 +26,126 @@
...
+      {{- if $objectData.pooler.acceptRO -}}
+        {{- $_ := set $objectData.pooler "type" "ro" -}}
+        {{- include "tc.v1.common.class.cnpg.pooler" (dict "rootCtx" $ "objectData" $objectData) -}}
+      {{- end -}}
+    {{- end -}}
+
+    {{- range $name, $backup := $objectData.backups.manual -}}
+      {{/* FIXME: I dont understand this here, we loop over manual backups
+      and each time we overwrite the $objectData.backupName etc.
+      So at the end backupName will have the name of the last manual backup. */}}
```
Comment (Ornias1993) [inline library/common/templates/spawner/_cnpg.tpl:85]: No, see the include...
After each pass, it creates a backup object, then it runs the loop again,

File: library/common/values.yaml [hunk 2]
```diff
@@ -754,36 +754,229 @@ cnpg:
...
+
+    # Cluster mode of operation. Available modes:
+    # * `standalone` - default mode. Creates new or updates an existing CNPG cluster.
+    # * `replica` - Creates a replica cluster from an existing CNPG cluster. # TODO
+    # * `recovery` - Same as standalone but creates a cluster from a backup, object store or via pg_basebackup.
+    mode: standalone
+
+
     database: "app"
     user: "app"
```
Comment (Ornias1993) [inline library/common/values.yaml:781]: both required

File: library/common/values.yaml [hunk 3]
```diff
@@ -754,36 +754,229 @@ cnpg:
...
+      # See: https://cloudnative-pg.io/documentation/current/api_reference/#bootstrapinitdb
+      initdb: {}
+        # postInitSQL:
+        #   - CREATE EXTENSION IF NOT EXISTS vector;
+
+      labels: {}
+      annotations: {}
+
+    # Recovery settings if the chosen mode is `recovery`.
+    recovery:
```
Comment (Ornias1993) [inline library/common/values.yaml:861]: only required when move is recovery

File: library/common-test/Chart.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@ appVersion: ""
 dependencies:
 - name: common
   repository: file://../common
-  version: ~14.2.0
+  version: ~15.0.0
```
Comment (Ornias1993) [inline library/common-test/Chart.yaml:6]: I dont think this is breaking, is it?

Comment (Ornias1993) [reply library/common-test/Chart.yaml:6]: well lets keep it at major anyway, that ensures users dont accidentally run into db breaking bugs again

Comment (stavros-k) [reply library/common-test/Chart.yaml:6]: Can't remember why I (?) had it marked as major.
But lets re-evaluate when finished

Comment (Ornias1993) [reply library/common-test/Chart.yaml:6]: I see now a few optional values changed place afaik, so technically breaking.

**Pull request merged**